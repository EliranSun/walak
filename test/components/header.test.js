import React from 'react';
import renderer from 'react-test-renderer';
import ShallowRenderer from 'react-test-renderer/shallow';
import Header from '../../src/components/header';

describe('header', () => {
  test('it should render a div', () => {
    const renderer = new ShallowRenderer();
    const component = renderer.render(<Header />);
    expect(component).toMatchSnapshot();
  });

  test('should render walak logo', () => {
    const component = renderer.create(<Header />);
    let tree = component.toJSON();
    expect(tree.children[0].children[0].type).toEqual('h1');
    expect(tree.children[0].children[0].children[0]).toEqual('וואלק');
    expect(tree.children[0].children[1].type).toEqual('h2');
    expect(tree.children[0].children[1].children[0]).toEqual('מה שעולה לראש');
  });

  test('should render walak menu', () => {
    const component = renderer.create(<Header />);
    let tree = component.toJSON();
    expect(tree.children[2].type).toEqual('ul');
    expect(tree.children[2].children[0].children[0]).toEqual('אודות');
    expect(tree.children[2].children[1].children[0]).toEqual('הכותבים');
  });

  test('should render search component', () => {
    const component = renderer.create(<Header />);
    let tree = component.toJSON();
    expect(tree.children[3].props.id).toEqual('search-component');
  });

  it('should pass needed props to Search component', () => {
    const component = renderer.create(<Header />);
    let tree = component.toTree();
    const propsNames = Object.keys(tree.rendered.rendered[3].props);
    expect(propsNames).toEqual(['articles', 'updateSearchResults']);
  });
});
