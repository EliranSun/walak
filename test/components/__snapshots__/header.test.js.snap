// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`header it should render a div 1`] = `
<div
  id="walak-header"
  style={Object {}}
>
  <div
    className="link"
    id="walak-logo"
  >
    <h1
      className="main-color"
      id="walak-title"
      style={Object {}}
    >
      וואלק
    </h1>
    <h2
      className="sub-color"
      id="walak-sub-title"
    >
      מה שעולה לראש
    </h2>
  </div>
  <hr />
  <ul
    className="secondary-color"
  >
    <li
      className="link"
    >
      אודות
    </li>
    <li
      className="link"
    >
      הכותבים
    </li>
  </ul>
  <Search />
</div>
`;
